File: python/CardSimulator.py
"""Simulates shuffling a standard deck of playing cards in a realistic manner.
"""

from CardDeck import CardDeck 


def main():
    deck = CardDeck()
    print(deck)
    print("")

    deck.shuffle()
    print(deck)
    deck.shuffle()
    print(deck)
    print("")

    deck.reset()
    print(deck)

if __name__ == "__main__":
    main()

__author__ = "Garrett Kamila Crayton (garrett.crayton@gmail.com)"
__date__ = "2021 April 11"
__credits__ = "Garrett Kamila Crayton (garrett.crayton@gmail.com)"
__version__ = "2.0"

File: PreviousVersions/v3/src/CardSimulator.java
/**
 * Simulates shuffling a deck of cards in a realistic fashion.
 *
 * @author Garrett Kamila Crayton
 * @version 1.0
 */
public class CardSimulator
{
    /**
     * Main class
     * @param args  C-style array declaration of parameter 'args'
     */
    public static void main(String[] args)
    {
        CardDeck testDeck = new CardDeck();

        System.out.println("Original: " + testDeck);
        testDeck.shuffle();
        System.out.println("Shuffled once: " + testDeck);
        testDeck.shuffle();
        System.out.println("Shuffled twice: " + testDeck);
        testDeck.reset();
        System.out.println("Re-sorted: " + testDeck);
    }
}